### ISSUE TITLE
Server no more available

### ISSUE THREAD

**Issue by picoroma** [jonwitts/nagios-speedtest#10]
Hi!
During the last period, I observed that the Servers available for the test is changing continuously.
So, if you writhe the server ID into a NAGIOS\ICINGA command, in case this server will change - the command does not work anymore.
Is possible to select the best server automatically?
As you do, when you run the script manually?
THX

**Comment by loopway** [comment]
I've had the same problem. Could solve it by updating speedtest-cli https://github.com/sivel/speedtest-cli to the latest version.

**Comment by loopway** [comment]
Update: unfortunately it did not solve it completely, but it happens less often now (might be coincidental).

We don't seem to be the only ones experiencing this:

- https://github.com/sivel/speedtest-cli/pull/763
- https://github.com/sivel/speedtest-cli/pull/765

**Comment by mavoga** [comment]
By analyzing the official test webpage requests it seems JSON server list at https://www.speedtest.net/api/js/servers?engine=js&limit=10&https_functional=true is used instead of XML at www.speedtest.net/speedtest-servers(-static)?.php and c.speedtest.net/speedtest-servers(-static)?.php.
The JSON list contains pretty accurate results, identical to website, while XML usually doesn't.
A ping analysis is then performed on each server in order to determine which one has the least latency and to get further info (IP, capabilities).

**Comment by jonwitts** [comment]
I found that the speedtest-cli from sivel was getting less and less reliable too. 

I have begun work on an updated plugin which uses the official binaries from Speedtest: https://www.speedtest.net/apps/cli

I have a version of this working at work; but I am not completely happy with how it runs. The speedtest-native branch of this repo has my code so far...

**Comment by mavoga** [comment]
Yes I had better to write on a sivel's pull request... the issue I'm experiencing is exactly the one you mentioned. Thanks, going to do that.

**Comment by borisv1979** [comment]
Just download version 1.3 of the script from https://github.com/iLem0n/nagios-speedtest/tree/0f0fd188a9ef0f37db71372ec3d6cd0784841c79 and use option for automatic server selection.

**Comment by TheITguy21** [comment]
Hi, not sure if you guys could make it work but I've reworked the old check_speedtest-cli Nagios plugin by John Witts to make it work with new Ookla SpeedTest binaries. If you guys want give it a try ;) https://github.com/TheITguy21/check_speedtest.sh
